lib/PipelineMemoryStream.cs

```diff
@@ -0,0 +1,158 @@
+﻿//------------------------------------------------------------------------------
+// <copyright file="PipelinedMemoryStream.cs" company="Microsoft">
+//    Copyright (c) Microsoft Corporation
+// </copyright>
+//------------------------------------------------------------------------------
+
+namespace Microsoft.WindowsAzure.Storage.DataMovement
+{
+    using System;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Collections.Generic;
+    using System.Threading;
+
+    // The following class is designed under the assumption that writes much are smaller on average than the buffer-size
```

Comment:
> writes much [](start = 64, length = 12)

Please optimize this description. Ensure the description is general, and can illustrate all your design goals.